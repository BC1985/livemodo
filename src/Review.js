import React, { Component } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";
import { faPortrait } from "@fortawesome/free-solid-svg-icons";
import "./BrowseReviewsPage.css";
import { parseDate } from "./utils/parseDate";

class Review extends Component {
  render() {
    const star = <FontAwesomeIcon icon={faStar} />;
    const userThumbnail = <FontAwesomeIcon icon={faPortrait} />;
    const review = this.props;
    return (
      <>
        <h2 className="band-name">{review.band_name}</h2>
        <p>Venue: {review.venue}</p>
        <p>Date:{parseDate(review.show_date)}</p>
        {star}
        <h4 className="tagline">{review.tagline}</h4>
        <p>posted: {parseDate(review.posted)}</p>
        <p className="user-name">
          <span className="user-thumbnail">{userThumbnail}</span> Username:
          {review.username}{" "}
        </p>
        <div className="review">{review.content}</div>
      </>
    );
  }
}

export default Review;
