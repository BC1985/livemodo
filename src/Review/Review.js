import React, { useEffect, useState } from "react"
import "../BrowseReviews/BrowseReviewsPage.css"
import { dateAndTime } from "../utils/parseDate"
import clsx from "clsx"
import Rating from "../Rating/Rating"
import { makeStyles } from "@material-ui/core/styles"
import { red } from "@material-ui/core/colors"
import Card from "@material-ui/core/Card"
import CardHeader from "@material-ui/core/CardHeader"
import Typography from "@material-ui/core/Typography"
import CardMedia from "@material-ui/core/CardMedia"
import CardContent from "@material-ui/core/CardContent"
import CardActions from "@material-ui/core/CardActions"
import Collapse from "@material-ui/core/Collapse"
import Avatar from "@material-ui/core/Avatar"
import IconButton from "@material-ui/core/IconButton"
import LocationOnOutlinedIcon from "@material-ui/icons/LocationOnOutlined"
import TodayOutlinedIcon from "@material-ui/icons/TodayOutlined"
import ExpandMoreIcon from "@material-ui/icons/ExpandMore"
import { arrayBufferToBase64 } from "../utils/imageUtil"
import config from "../config"

function Review({ review }) {
  const [expanded, setExpanded] = useState(false);
  const [image, setImage] = useState("")
  const [avatar, setAvatar] = useState("")
  const { content, tagline, venue, rating, createdAt, bandName, showDate } =
    review
  const getImage = async () => {
    try {
      const data = await fetch(`${config.API_BASE_URL}/images`)
      const img = await data.json()
      const base64Flag = "data:image/jpeg;base64,"
      const imageStr = arrayBufferToBase64(img.img.data.data)
      setImage(base64Flag + imageStr)
    } catch (e) {
      console.log(e)
    }
  };
  const getAvatar = async () => {
    try {
      const data = await fetch(`${config.API_BASE_URL}/images/avatars`)      
      const avatar = await data.json();
      const base64Flag = "data:image/jpeg;base64,";
      const imageStr = arrayBufferToBase64(avatar.img.data.data);
      setAvatar(base64Flag + imageStr);
      console.log('AVATAR',avatar)
    } catch (e) {
      console.log(e);
    }
  };
  useEffect(() => {
    getAvatar()
    getImage()
  }, [])

  const useStyles = makeStyles(theme => ({
    root: {
      maxWidth: 345,
      paddingBottom: 100,
    },
    media: {
      height: 200,
    },
    expand: {
      transform: "rotate(0deg)",
      marginLeft: "auto",
      transition: theme.transitions.create("transform", {
        duration: theme.transitions.duration.shortest,
      }),
    },
    card: {
      marginBottom: "10%",
    },
    icon: {
      marginRight: "10px",
    },
    expandOpen: {
      transform: "rotate(180deg)",
    },
  }))
  const classes = useStyles()

  const handleExpandClick = () => {
    setExpanded(!expanded)
  }

  return (
    <>
      <Card className={classes.card}>
        <CardHeader
          avatar={
            <Avatar
              aria-label="avatar"
              variant="rounded"
              alt="user"
              src={avatar}
            >
            </Avatar>
          }
          title={tagline}
          subheader={dateAndTime(createdAt, "PPPPp")}
        />
        <CardMedia
          component="img"
          src={image}
          className={classes.media}
        />

        <CardContent>
          <p className="user-name"></p>
          <Typography variant="h4" component="h5">
            {bandName}
          </Typography>
          <Typography variant="h5">
            <LocationOnOutlinedIcon className={classes.icon} />
            {venue}
          </Typography>
          <Typography variant="subtitle1">
            <TodayOutlinedIcon className={classes.icon} />
            {dateAndTime(showDate, "PPPP")}
          </Typography>
          <Rating value={rating} />
          <CardActions disableSpacing>
            <IconButton
              className={clsx(classes.expand, {
                [classes.expandOpen]: expanded,
              })}
              onClick={handleExpandClick}
              aria-expanded={expanded}
              aria-label="show more"
            >
              <ExpandMoreIcon />
            </IconButton>
          </CardActions>
        </CardContent>
        <Collapse in={expanded} timeout="auto" unmountOnExit>
          <CardContent>
            <Typography variant="body1">{content}</Typography>
          </CardContent>
        </Collapse>
      </Card>
    </>
  )
}

export default Review
