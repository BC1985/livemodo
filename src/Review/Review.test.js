import React from "react";
import ReactDOM from "react-dom";
import Review from "../Review/Review";
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Review />, div);
  ReactDOM.unmountComponentAtNode(div);
});
