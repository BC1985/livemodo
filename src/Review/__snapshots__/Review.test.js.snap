// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="review"
>
  <p
    className="user-name"
  >
    <span
      className="user-thumbnail"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-portrait fa-w-12 "
        data-icon="portrait"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M336 0H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zM192 128c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H102.4C90 384 80 375.4 80 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
     
    <span
      className="username"
    >
       
    </span>
  </p>
  <p>
    posted: 
    NaN/NaN/NaN
  </p>
  <h2
    className="band-name"
  />
  <h4
    className="tagline"
  />
  <p>
    Venue: 
  </p>
  <p>
    Date: 
    NaN/NaN/NaN
  </p>
  <div
    className="rating"
  >
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
  <div
    className="review-content"
  />
</div>
`;
